cfme/utils/log_validator.py

```diff
@@ -77,12 +78,22 @@ def _check_match_logs(self, line):
         for pattern in self.matched_patterns:
             if re.search(pattern, line):
                 logger.info('Expected pattern {} was matched on line {}'.format(pattern, line))
-                self.matches[pattern] = True
+                self.matches[pattern] = self.matches[pattern] + 1
 
-    def _verify_match_logs(self):
-        for pattern in self.matched_patterns:
-            if pattern not in self.matches:
-                pytest.fail('Expected pattern {} did not match'.format(pattern))
+    def validate_logs(self):
+        """Validate log pattern"""
+        self._validation()
+        for pattern, count in self.matches.items():
+            if count == 0:
+                pytest.fail(
```

Comment:
I think this is a good opporunity to 'fix' this behavior, and not use pytest outside of a pytest context. I think we should instead be raising an `AssertionError` here

Reply:
:+1: Here

Reply:
If we like to avoid `pytest context` in such utilities. As per me, `AssertionError` is a bad idea. what do you think Mike? if method name should `validate` only with returning `bool` (True/ False).

Reply:
Leaving this behavior as-is for now, will update in further enhancement PR for LogValidator where callers are updated.